content/docs/start/experiments/building-pipelines.md

```diff
@@ -0,0 +1,208 @@
+
+Once you start consolidating your project and the code structure, the
+flexibility of the notebooks might start to lose its value and some parts of the
+workflow could be improved.
+
+DVC <abbr>Pipelines</abbr> can help you standardize your workflow following
+software engineering best practices while also enabling new features like:
+
+- [Preventing steps to be recomputed](/doc/user-guide/pipelines/run-cache).
+- [Modifying parameters from the CLI](/doc/user-guide/experiment-management/running-experiments#tuning-hyperparameters).
+- [Queueing and executing multiple experiments at once](/doc/user-guide/experiment-management/running-experiments#the-experiments-queue).
+- [Configuring your project with Hydra](/doc/user-guide/experiment-management/running-experiments#hydra-composition).
+
+## From notebook to scripts
+
+We are not going to cover here how to go from notebooks to scripts, but you can
+follow
+[this detailed blog post](https://towardsdatascience.com/from-jupyter-notebook-to-sc-582978d3c0c)
+to learn how to do it.
+
+From a high level perspective, what you need to do before building a pipeline
+is:
+
+- **Modularize**: split the different logical steps in your notebook into
+  separate scripts.
+
+In our example repo, the
+[original notebook](https://github.com/iterative/example-get-started-experiments/blob/main/notebooks/TrainSegModel.ipynb)
+has been transformed into 3 scripts:
+[`data_split.py`](https://github.com/iterative/example-get-started-experiments/blob/main/src/data_split.py),
+[`train.py`](https://github.com/iterative/example-get-started-experiments/blob/main/src/train.py)
+and
+[`evaluate.py`](https://github.com/iterative/example-get-started-experiments/blob/main/src/evaluate.py).
+
+- **Parametrize**: adapt your scripts to decouple the configuration from the
+  source code.
+
+All these scripts are configured using different sections of the
+[`params.yaml`](https://github.com/iterative/example-get-started-experiments/blob/main/params.yaml)
+file.
+
+## Creating stages
+
+You can use `dvc stage add` to transform a script into a <abbr>stage</abbr>:
+
+```cli
+$ dvc stage add -n data_split \
+  -p base,data_split \
+  -d data/pool_data -d src/data_split.py \
+  -o data/train_data -o data/test_data \
+  python src/data_split.py
+```
+
+A `dvc.yaml` file is generated. It includes information about the command you
+want to run (`python src/data_split.py`), its <abbr>dependencies</abbr>,
+<abbr>parameters</abbr>, and <abbr>outputs</abbr>:
+
+```yaml
+stages:
+  prepare:
+    cmd: python src/data_split.py
+    deps:
+      - src/data_split.py
+      - data/pool_data
+    params:
+      - base
+      - data_split
+    outs:
+      - data/train_data
+      - data/test_data
+```
+
+DVC uses these metafiles to track the data used and produced by the stage, so
+there's no need to use `dvc add` on `data/train_data` or `data/test_data`
+[manually](/doc/start/data-management/data-versioning).
+
+## Building a DAG
```

Comment:
> Realistically, I think anyone would want to run the first stage before adding more. I think it also highlights the incremental approach of DVC pipelines, which makes them less intimidating. It feels like it's trying to save `dvc exp run` for the next section rather than focusing on a realistic workflow.

Could you expand a little more on the reasoning?
I don't really understand why running the unfinished pipeline would make it less intimidating and why it would be a more realistic workflow.